src/app/WriteHandler.cpp

```diff
@@ -460,12 +480,11 @@ CHIP_ERROR WriteHandler::ProcessGroupAttributeDataIBs(TLV::TLVReader & aAttribut
             // Try to get the metadata from for the attribute from one of the expanded endpoints (it doesn't really matter which
             // endpoint we pick, as long as it's valid) and update the path info according to it and recheck if we need to report
             // list write begin.
-            if (attributeMetadata == nullptr)
+            if (needToCheckListOperation)
             {
-                attributeMetadata = GetAttributeMetadata(dataAttributePath);
-                bool currentAttributeIsList =
-                    (attributeMetadata != nullptr && attributeMetadata->attributeType == kListAttributeType);
-                if (!dataAttributePath.IsListOperation() && currentAttributeIsList)
+                needToCheckListOperation = false;
+
+                if (!dataAttributePath.IsListOperation() && IsListAttributePath(dataAttributePath))
```

Comment:
This change is wrong.  If the attribute does not exist on `mapping.endpoint_id`, this will produce a garbage value for `IsListAttributePath` and never check again, so when it gets to endpoints that _do_ have the attribute the right things won't happen.

Note that the old code did not have this problem: if it go a null from GetAttributeMetadata it would avoid munging the path for this endpoint, but then obviously fail out of the write anyway, so that would not matter much, and for later endpoints it would try to get the metadata again.

Reply:
We are missing test coverage :(